controllers/loadtest_controller_test.go

```diff
@@ -502,3 +501,134 @@ var _ = Describe("Pod Creation", func() {
 		})
 	})
 })
+
+var _ = Describe("CheckMissingPods", func() {
+
+	var currentLoadTest = newLoadTestWithMultipleClientsAndServers() // nothing modify this --> okay
+	var expectedReturnList []*grpcv1.Component                       //initialized every time after a test -->okay
+	var allRunningPods = &corev1.PodList{Items: []corev1.Pod{}}
+	var returnedList []*grpcv1.Component // it was a newlist generated form myfun everytime, address are different --> okay
+
+	JustBeforeEach(func() {
+		returnedList = CheckMissingPods(currentLoadTest, allRunningPods)
+	})
+
+	AfterEach(func() {
+		expectedReturnList = []*grpcv1.Component{}
+		newThing := corev1.PodList{Items: []corev1.Pod{}}
+		allRunningPods = &newThing
+	})
+
+	Describe("no pods from current loadtest is running", func() {
+		BeforeEach(func() {
+			//Set up the expected returned list, in these cases should be full list
+			for i := 0; i < len(currentLoadTest.Spec.Clients); i++ {
+				expectedReturnList = append(expectedReturnList, &currentLoadTest.Spec.Clients[i].Component)
+			}
+			for i := 0; i < len(currentLoadTest.Spec.Servers); i++ {
+				expectedReturnList = append(expectedReturnList, &currentLoadTest.Spec.Servers[i].Component)
+			}
+			expectedReturnList = append(expectedReturnList, &currentLoadTest.Spec.Driver.Component)
+		})
+
+		Context("allRunningPods is empty", func() {
+			allRunningPods = &corev1.PodList{Items: []corev1.Pod{}}
+			It("return the full list", func() {
+				Expect(checkIfEqual(returnedList, expectedReturnList)).To(Equal(true))
```

Comment:
The `checkIfEqual` helper is okay for now. Let's prefer the `BeTrue()` matchers, like: 

```suggestion
				Expect(checkIfEqual(returnedList, expectedReturnList)).To(BeTrue())
```

fun experiment: If you really wanted to, you could create your own matcher, like `Expect(returnedList).To(ContainEquivalentNamesToList(expectedReturnList))`. See [GoMega custom matchers](https://onsi.github.io/gomega/#adding-your-own-matchers). I have never done this.

Reply:
Great! I definitely should look more into the provided matchers!